Disable KSCAN in MCHP boards to fix obj_tracing test

franciscomunoz opened the pull request:
This change addresses issue# 21756 where the expected thread
count does not match because kscan module is on in our boards.

nashif reviewed (changes requested):
this is the wrong fix. Platforms executing the kernel tests should not enable any drivers or subsystems beside those needed to boot up the system. Kernel tests make lots of assumptions about the threads and objects being used and such test may fail if the platform brings in more objects.
Ideally this should be fixed by having the platform defconfig just have the bare minimum of kconfigs enabled, in your application you can then enable anything you need.

franciscomunoz committed:
boards: mec15xxevb_assy6853: Disable KSCAN to avoid test case issues

This change addresses issue# 21756 where the expected thread
count does not match because kscan module is enabled in this board.
The Microchip kscan driver is contributing to the total thread count,
therefore interfering with the expected test result.

Signed-off-by: Francisco Munoz <francisco.munoz.ruiz@intel.com>

franciscomunoz committed:
boards: mec1501modular_assy6885: Disable KSCAN to avoid test case issues

This change also addresses issue# 21756 tests/obj_tracin where the
expected thread count does not match because kscan module is enabled
in this board. The Microchip kscan driver is contributing to the total
thread count, therefore interfering with the expected test result.

Signed-off-by: Francisco Munoz <francisco.munoz.ruiz@intel.com>
